inc/class-core-sitemaps.php

```diff
@@ -110,11 +111,16 @@ public function register_rewrites() {
 		add_rewrite_rule( '^wp-sitemap-index\.xsl$', 'index.php?sitemap-stylesheet=index', 'top' );
 
 		// Register routes for providers.
-		$providers = core_sitemaps_get_sitemaps();
-
-		foreach ( $providers as $provider ) {
-			add_rewrite_rule( $provider->route, $provider->rewrite_query(), 'top' );
-		}
+		add_rewrite_rule(
+			'^wp-sitemap-([a-z]+?)-([a-z-\d]+?)-([0-9]+?)\.xml$',
```

Comment:
Minor suggestion for the `sub-type` part of the regex, as stated in the [PHP manual](https://www.php.net/manual/en/regexp.reference.character-classes.php):

> If a minus character is required in a class, it must be escaped with a backslash or appear in a position where it cannot be interpreted as indicating a range, typically as the first or last character in the class. 

So, I think the regex would be more clear/unambiguous if it were:

`^wp-sitemap-([a-z]+?)-([a-z\d-]+?)-([0-9]+?)\.xml$`

or

`^wp-sitemap-([a-z]+?)-([-a-z\d]+?)-([0-9]+?)\.xml$`

I usually use the later, as I think it's easier to see the `-` when it's at the beginning of the character class.

Reply:
Sounds good!

I'll also replace `[0-9]` at the end with `[\d]` while I'm at it.